### ISSUE TITLE
Seperate runtime data from config file

### ISSUE THREAD

**Issue by DonRichie** [quodlibet/quodlibet#2989]
Hi, 

I am currently using a dotfile repository in which I included ~/.config/quodlibet/config and my playlists.
In the mentioned configuration file quodlibet remembers the last song played in a line named "song = ..." . There are some more lines changing on every program execution like for example "quodlibet_position = ...".

My dotfile management recognizes those changes and shows me an unclean state.

Could you do me a favour and outsource that temporary data in a seperate file? This way I could ignore the temporary file and only need a new commit when something important changes.

**Comment by DonRichie** [comment]
I clicked some button in the program which resulted in the following diff:

```
diffing "f_quodlibet_config" VS "~/.config/quodlibet/config"
9c9
< browser = Paned
---
> browser = InternetRadio
42c42
< quodlibet_position = 138 202
---
> quodlibet_position = 2197 67
51c51
< song = /path/to/currently/playing/file.opus
---
> song = 
64c64
< audiofeeds_pos = 0.2561349693251534
---
> audiofeeds_pos = 0.2563338301043219
69,71c69,71
< internetradio_pos = 0.40017211703958694
< pane_selection = 0	.
< 	0	.
---
> internetradio_pos = 0.40014903129657226
> pane_selection = 1	.
> 	1	.
78a79
> radio = indie
87,91c88
< active_plugins = Search artist in Wikipedia
< 	WaveformSeekBar
< 	Playlist Export
< 	Tray Icon
< 	exact-rating
---
> active_plugins = Playlist Export
93,94c90
< 	Website Search
< 	mpris
---
> 	exact-rating
95a92
> 	Tray Icon
96a94,95
> 	Website Search
> 	WaveformSeekBar
97a97,98
> 	Search artist in Wikipedia
> 	mpris
```

Unfortunately I don't know enough about the config file to make you a full list of the relevant settings.

**Comment by frestr** [comment]
Most of settings in the config file is about the UI state and aren't really "temporary". It's just that they change whenever you change something in the UI (some noise is generated by floating point imprecision though).

IMO it doesn't make as much sense to use a git repo for config files you don't edit by hand. Separating values set in the global/browser/plugin preferences etc. and values generated otherwise would add what I see as unneeded complexity.

**Comment by DonRichie** [comment]
I agree that temporary data is the wrong term. The data is stored in the file to make it persistent until the next program start.

It makes sense to include any config file in the git repository regardless of the way how it is edited (By hand or by a GUI dialog). The point is to easily share the configuration for programs with different devices.

**Comment by ohel** [comment]
I kind of see the point, but at the same time it's a file probably not updated that often (except for the "temporary data" parts obviously) and especially most of the configurations I think are usually system-dependent: for example, equalizer plugin for one system, crossfeed plugin for the other. Looking at my own config file, there isn't that much info I'd like to transfer between systems - and if there is, I pretty much never change it anyway. The column and ReplayGain settings are pretty much the only things constant I have. I do have my Quod Libet output pipeline config (it's a long one with a parametric eq) stored in a separate file in my dotfiles repo. But I update it maybe once a year so it's fine as it is.

With that in mind, how about using a stripped config file for the dotfiles repo (like I do with my pipeline config), and for each new system just copy the contents from the stripped config file to the real system-specific config file? Or what exactly is the use case here?

**Comment by lazka** [comment]
I can see the point, but I'm not sure this use case is worth changing things.

We could add a second config instance called something like "state" and write things there instead.
